project-base/storefront/helpers/getInternationalizedStaticUrls.ts

```diff
@@ -1,29 +1,32 @@
+import { STATIC_REWRITE_PATHS, StaticRewritePathKeyType } from 'config/staticRewritePaths';
 import {
     getQueryWithoutSlugTypeParameterFromQueryString,
     getUrlWithoutGetParameters,
 } from 'helpers/parsing/urlParsing';
 import { GetServerSidePropsContext, NextPageContext } from 'next';
-import getConfig from 'next/config';
+import { SameLengthOutput } from 'types/SameLengthOutput';
 
-type Url = string | { url: string; param: string | undefined | null };
+type Url = StaticRewritePathKeyType | { url: StaticRewritePathKeyType; param: string | undefined | null };
 
 const getInternationalizedStaticUrl = (url: Url, domainUrl: string) => {
-    const { publicRuntimeConfig } = getConfig();
-    const urlsOnDomain = publicRuntimeConfig.staticRewritePaths[domainUrl];
+    const urlsOnDomain = STATIC_REWRITE_PATHS[domainUrl];
 
     if (typeof url === 'string') {
         const result = urlsOnDomain[url];
         return typeof result !== 'undefined' ? result : '';
     }
 
     const staticUrlTemplate = urlsOnDomain[url.url];
-    const staticPart = staticUrlTemplate?.split(':')[0];
+    const staticPart = staticUrlTemplate.split(':')[0];
 
-    return (staticPart ?? '') + (url.param ?? '');
+    return staticPart + (url.param ?? '');
 };
 
-export const getInternationalizedStaticUrls = (urls: Url[], domainUrl: string): string[] => {
-    return urls.map((url) => getInternationalizedStaticUrl(url, domainUrl));
+export const getInternationalizedStaticUrls = <InputUrls extends Url[]>(
+    urls: [...InputUrls],
+    domainUrl: string,
+): SameLengthOutput<InputUrls> => {
```

Comment:
Since we agreed that we won't use return type for places where it doesn't have any benefit, let's remove return type. Especially in this place where return is with `as`.

Reply:
Removed, thanks for noticing